### DISCUSSION TITLE
Using ITelemetryInitializer with Durable Entities

### DISCUSSION THREAD

**Discussion by NattyMojo** [Azure/azure-functions-durable-extension#2409] [Q&A]
I haven't seen too many people talking about this so trying to find some answers as to what I'm seeing here. I've used ITelemetryInitializer before and have a class `ExceptionDataInitializer: ITelemetryInitializer` that I'm registering in the Startup of my .NET 6 Azure Durable Function application like so `builder.Services.AddSingleton<ITelemetryInitializer, ExceptionDataInitializer>();`

I've done this with other projects (without durable functions/entities) and it works great to be able to add data into the `customDimensions` section of a record in the Azure Application Insights for any Exception that happens. But, for a new project we've decided to give Durable Entities a try, and want to use this same ExceptionDataInitializer. One thing I immediately noticed when trying it was that the properties we'd expect in the exception logs, are not there. We know that using `SignalEntityAsync` puts the handler into a queue that Durable Functions then handles in the background, but it seems to not be using the registered initializer when it runs into exceptions and sends them out to application insights. 

Is there something I'm missing here? I would assume that the runtime for the Durable Functions/Entities would still be using the registered services from the Startup.

Thanks :)

**Comment by davidmrdavid** [comment]
@bachuv, @sebastianburckhardt: is this something you two would know about?

**Reply by drdamour** [reply]
any updates on this?

**Reply by bachuv** [reply]
@NattyMojo, the registered services should get picked up. Can you share a minimal repro that we could take a look at? It would help us investigate the issue quickly. Thanks!

**Reply by NattyMojo** [reply]
@bachuv while putting a minimal repo together I did some extra testing to sanity check myself and found that we were just wrong about Exception nesting for the errors. Our `ExceptionDataInitializer` was gathering the top-level exception because when we've done this in the past we were able to catch the exceptions and do a `ILogger.LogError()` which works great because the exception being handled is the only one we threw. Turns out that the runtime wraps the ones thrown in a durable entity signaler twice and we didn't know that until I did the extensive testing. No issue here, we just need to fix out ITelemetryInitializer implementation :)

**Reply by bachuv** [reply]
@NattyMojo, thanks for the update!
